core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2715,6 +2715,66 @@ private[spark] object Utils extends Logging {
     HashCodes.fromBytes(secretBytes).toString()
   }
 
+  /**
+   * A safer version than scala obj's getClass.getSimpleName and Utils.getFormattedClassName
+   * which may throw Malformed class name error.
+   * This method mimicks scalatest's getSimpleNameOfAnObjectsClass.
+   */
+  def getSimpleName(fullyQualifiedName: String): String = {
+    stripDollars(parseSimpleName(fullyQualifiedName))
+  }
+
+  /**
+   * Remove the packages from full qualified class name
+   */
+  private def parseSimpleName(fullyQualifiedName: String): String = {
```

Comment:
`fullyQualifiedName.split("\\.").takeRight(1)`?

Reply:
this method was copied from scalatest. I think it makes sense to update in this cleaner way